hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/AbstractAbfsIntegrationTest.java

```diff
@@ -552,16 +552,7 @@ protected void assumeValidAuthConfigsPresent() {
         currentAuthType == AuthType.SAS);
     if (currentAuthType == AuthType.SharedKey) {
       assumeValidTestConfigPresent(getRawConfiguration(), FS_AZURE_ACCOUNT_KEY);
-    } else if (currentAuthType == AuthType.OAuth) {
```

Comment:
Why is this removed ?

Reply:
Removed code was to make sure that required configs to work with OAuth are present. This was a redundant check as a similar check is already present in AbfsConfiguration::getTokenProvider() method where these fields are parsed as Mandatory fields. getMandatoryPasswordField will take care of throwing relevant exception in case these are not configured.

Moreover the redundant check was working only for Client-Secret Token Provider and not for other OAuth Token Providers.

Reply:
@anujmodi2021 : Can you pls give a try to add the `testConfigPropNotFound` kinda unit test cases for the Oauth configs to make sure the source code validations are respected. Say, if someone tries to refactor getMandatoryPasswordString() to getPasswordString() function unknowingly, these unit test cases can make sure the behavior is retained by failing the test, isn't it?

```
          String authority = appendSlashIfNeeded(
              getTrimmedPasswordString(FS_AZURE_ACCOUNT_OAUTH_MSI_AUTHORITY,
              AuthConfigurations.DEFAULT_FS_AZURE_ACCOUNT_OAUTH_MSI_AUTHORITY));
          String tenantGuid =
              getMandatoryPasswordString(FS_AZURE_ACCOUNT_OAUTH_MSI_TENANT);
          String clientId =
              getMandatoryPasswordString(FS_AZURE_ACCOUNT_OAUTH_CLIENT_ID);
          String tokenFile =
              getTrimmedPasswordString(FS_AZURE_ACCOUNT_OAUTH_TOKEN_FILE,
              AuthConfigurations.DEFAULT_FS_AZURE_ACCOUNT_OAUTH_TOKEN_FILE);
```

Probably you can write the test cases into this test class.
https://github.com/apache/hadoop/blob/trunk/hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/TestAccountConfiguration.java#L377


Secondly, I could see some interesting test cases `testGlobalAndAccountOAuthPrecedence`. Can you pls go through the class and add necessary test cases.
https://github.com/apache/hadoop/blob/trunk/hadoop-tools/hadoop-azure/src/test/java/org/apache/hadoop/fs/azurebfs/TestAccountConfiguration.java#L405

Reply:
Thanks for the suggestion Rakesh..
I see that this test is already present in the class that you pointed to. It is there for testing ClientSecret Based OAuth configs. I will add similar for other OAuth token providers as well including Workload Identity Token Provider.